### PR TITLE
SingleApplication: Fix double free on exit

### PR DESCRIPTION
**Pull request by jonaski**
The private data is free'd in the destructor.

### CODE REVIEW

File: singleapplication.cpp [hunk 1]
```diff
@@ -158,8 +158,6 @@ SingleApplication::SingleApplication( int &argc, char *argv[], bool allowSeconda
 
     d->connectToPrimary( timeout, SingleApplicationPrivate::NewInstance );
 
-    delete d;
```
Comment (Shatur) [inline singleapplication.cpp:161]: It's constructor, destructor won't be called in this case.

File: singleapplication.cpp [hunk 2]
```diff
@@ -262,7 +260,6 @@ void SingleApplication::abortSafely()
     Q_D( SingleApplication );
 
     qCritical() << "SingleApplication: " << d->memory->error() << d->memory->errorString();
-    delete d;
```
Comment (Shatur) [inline singleapplication.cpp:265]: This one is good.

**Comment by itay-grudev** [comment]
@jonaski Is this causing a crash for you - like a non-zero exit code?

**Review by Shatur — APPROVED** [review]

**Comment by jonaski** [comment]
Nevermind, abortSafely is only called from the constructor.

**Pull request closed without merging**